### DISCUSSION TITLE
SAP Predictive Analytics - JRI interface issue

### DISCUSSION THREAD

**Discussion by aaronkrech** [s-u/rJava#278] [Q&A] [answered]
Hello,

I must use SAP Predictive Analytics for school.  I am installing SAP.PA version 3.3.7, which requires R version 3.4.  I am having a JRI interface issue when trying to run an R-linear regression.  I have traced the issue to not having the rJava package in my R library.  When I try to install via Packages>install packages in R, rJava is not in the list.  I have tried to install version 1.0-1 with rJava_1.0-1.tar.jz also with Packages>Install packages from local files and I get an error stating rJava requires R version 3.6 or higher.  I have tried using the newer version of R and integrating it into SAP.PA to no avail.  I also attempted to install rJava_0.9-7.tar.jz and that installation fails also.

Here is the text from the 0.9-7 installation failure:
> utils:::menuInstallLocal()
* installing *binary* package 'rJava' ...
Warning: running command 'mv "C:/Users/Public/R-3.4.0/library/rJava" "C:/Users/Public/R-3.4.0/library/00LOCK-rJava/rJava"' had status 127
Warning: running command 'cp -R . "C:/Users/Public/R-3.4.0/library/rJava" || ( tar cd - .| (cd "C:/Users/Public/R-3.4.0/library/rJava" && tar -xf -))' had status 127
ERROR: installing binary package failed
* removing 'C:/Users/Public/R-3.4.0/library/rJava'
Warning messages:
1: running command '"C:/Users/Public/R-3.4.0/bin/x64/R" CMD INSTALL -l "C:\Users\Public\R-3.4.0\library" "C:/Users/AKREC/Downloads/rJava_0.9-7.tar.gz"' had status 1 
2: In install.packages(files[tarballs], .libPaths()[1L], repos = NULL,  :
  installation of package ‘C:/Users/AKREC/Downloads/rJava_0.9-7.tar.gz’ had non-zero exit status

Does anyone have any thoughts?  I'm hoping SAP hasn't updated Predictive Analytics past a useable R version.
Thank you

**Comment by s-u** [comment, marked as answer]
@aaronkrech it looks like a permission issue - you don't seem to have permission to install packages in the library you are using. You should install it into your own personal library instead.

You really want to use the CRAN binary version, because compiling Java from sources is non-trivial on Windows (you need tools, JDK, etc.) - and it is not necessary.

Microsoft provides an archive of binaries [MRAN Timemachine](https://mran.microsoft.com/timemachine), so for R 3.4.4 the following was the binary at the time which you could try:

https://cran.microsoft.com/snapshot/2018-04-01/bin/windows/contrib/3.4/rJava_0.9-9.zip

Note that you still have to fix your permission issues fist. You can also use the above as a mirror by using

```r
install.packages("rJava",  repos="https://cran.microsoft.com/snapshot/2018-04-01")
```

**Reply by aaronkrech** [reply]
@s-u 

Dude, you're a genius!  I ran R as administrator, pasted in the install command you posted and rJava successfully installed.  What's even better is the JRI from SAP.PA works.  Now my entire class can do their homework!

Thanks for your help!

**Reply by s-u** [reply]
Great, I'm glad it worked :)

**Comment by nahtajainsurbhi** [comment]
Hi,
I am also having the same issue and found the best answer from [SAP Courses](https://sitefeatures.ideas.aha.io/ideas/LAWEB-I-482).
